src/sinks/pulsar/config.rs

```diff
@@ -171,6 +176,47 @@ pub enum PulsarCompression {
     Snappy,
 }
 
+#[configurable_component]
+#[configurable(description = "Custom connection retry options configuration for the Pulsar client.")]
+#[derive(Clone, Debug)]
+pub struct CustomConnectionRetryOptions {
+    /// Minimum delay between connection retries.
+    #[configurable(metadata(docs::type_unit = "milliseconds"))]
+    #[configurable(metadata(docs::examples = 10))]
+    pub min_backoff: Duration,
```

Comment:
```suggestion
    pub min_backoff_ms: Duration,
```

We generally suffix duration types with their unit per https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#option-naming

Reply:
https://github.com/vectordotdev/vector/pull/21245/commits/18c535dfe89382f547d0a700bf38e4eb350f1c39